sbi/inference/posteriors/mcmc_posterior.py

```diff
@@ -230,7 +230,8 @@ def sample(
 
         init_fn = self._build_mcmc_init_fn(
             self.proposal,
-            self.potential_fn,
+            # Use transformed potential for sampling inits.
+            self.potential_,
```

Comment:
Good catch, I agree that this is a bug. 

However, I think that this fix leads to problems: The `self.potential_` is the final potential, i.e. it also takes care of being `pyro` consistent. In other words: if the user wants to use a `pyro` mcmc sampler, the `self.potential_` expects a `Dict` as input (not a `Tensor`). This would break in `init_strategy.py`, because the `self.proposal` always returns a `Tensor`.

Instead I suggest a different fix: [here](https://github.com/mackelab/sbi/blob/8d78b93ad85acbfeac413adf963db24c23939cda/sbi/samplers/mcmc/init_strategy.py#L66-L68), we compute the `log_weights` before transforming the samples.

Maybe I'm missing something though.

Reply:
good catch. 
yes, I thought about fixing it there as well. I was wondering why we are performing `sir` in transformed space anyways. I think the reason is that we want transformed init samples. Thus, we could just perform `sir` in untransformed space and then transform the init samples on the return (like we do it for the `proposal` strategy), right?

Reply:
yes, exactly!